tests/httpClient/deduplicateRequestAdapter.test.ts

```diff
@@ -0,0 +1,26 @@
+import { AxiosRequestConfig } from 'axios';
+import { createDeduplicateRequestAdapter } from '../../src/httpClient/deduplicateRequestAdapter';
+
+describe('deduplicateRequestAdapter()', () => {
+  it('deduplicates calls to the passed adapter', async () => {
+    const mockAdapter = jest.fn().mockResolvedValue({});
+    const mockDeduplicationKeyProvider = jest.fn().mockReturnValue('testDedupId');
+
+    const dedupAdapter = createDeduplicateRequestAdapter(mockAdapter, mockDeduplicationKeyProvider);
+    const request: AxiosRequestConfig = {
+      url: 'testUrl',
+    };
+
+    // fire two requests in parallel, only 1 should be made to the underlying adapter
+    dedupAdapter(request);
```

Comment:
[Optional] I think we could add some `setTimeout` inside the mockAdapter. But if it really works consistently like this, never mind

Reply:
Ok, added 10ms to simulate async behavior. Now I see that I should also `await dedupAdapter(request)` otherwise there is no guarantee it actually finished before moving to expectations.

Have a look, I've changed the test quite a bit.